wordpress-mobile/WordPress-iOS
# Issue/5245 migrate account service to nsurlsession

Refs #5245 

To test: 
 - Login in the app using the regular login flow/ (user/password)
 - Check if the blog list, blog visibility and user account details are retrieved correctly
 - Login in the app using the new login flow (email then magic link)

Needs review: @koke can you have a more global look and @aerych  do you mind to check the code for magic links to see if I miss something there.
